File: TheaterSuggestions/Java/SeatingPlaceSuggestionsAcceptanceTests/src/test/java/org/weaveit/SeatsSuggestionsAcceptanceTests/SeatingPlace.java
package org.weaveit.SeatsSuggestionsAcceptanceTests;

public class SeatingPlace {

    private String rowName;
    private int number;
    private PricingCategory pricingCategory;
    private SeatingPlaceAvailability seatingPlaceAvailability;

    public SeatingPlace(String rowName, int number, PricingCategory pricingCategory, SeatingPlaceAvailability seatingPlaceAvailability) {
        this.rowName = rowName;
        this.number = number;
        this.pricingCategory = pricingCategory;
        this.seatingPlaceAvailability = seatingPlaceAvailability;
    }

    public boolean isAvailable() {
        return seatingPlaceAvailability == SeatingPlaceAvailability.Available;
    }


    public void updateCategory(SeatingPlaceAvailability seatingPlaceAvailability) {
        this.seatingPlaceAvailability = seatingPlaceAvailability;
    }

    @Override
    public String toString() {
        return rowName + number;
    }
}

File: TheaterSuggestions/CSharp/TheaterSuggestions.Tests/SeatingPlace.cs
namespace SeatsSuggestions.Tests;

public class SeatingPlace
{
    public SeatingPlace(string rowName, uint number, PricingCategory pricingCategory, SeatingPlaceAvailability seatingPlaceAvailability)
    {
        RowName = rowName;
        Number = number;
        PricingCategory = pricingCategory;
        SeatingPlaceAvailability = seatingPlaceAvailability;
    }

    public string RowName { get; }
    public uint Number { get; }
    public PricingCategory PricingCategory { get; }
    private SeatingPlaceAvailability SeatingPlaceAvailability { get; set; }

    public bool IsAvailable => SeatingPlaceAvailability == SeatingPlaceAvailability.Available;

    public override string ToString()
    {
        return $"{RowName}{Number}";
    }

    public void UpdateCategory(SeatingPlaceAvailability seatingPlaceAvailability)
    {
        SeatingPlaceAvailability = seatingPlaceAvailability;
    }
}